java/src/main/java/com/bazaarvoice/gumshoe/Gumshoe.java

```diff
@@ -98,11 +100,12 @@ public Context context(String name) {
         Context context;
 
         if (contexts.isEmpty()) {
-            context = new Context(name, eventFactory, eventHandler);
+            context = new Context(this, name, eventFactory, eventHandler);
         } else {
             context = new Context(name, contexts.peek());
         }
         contexts.push(context);
+        ensureContextStackDepthNotExceeeded();
```

Comment:
I'd prefer if we did this check at the beginning.  It feels better to me to know that if this exception is thrown, the original state is unchanged.